samples/boot/saml2login/src/main/java/boot/saml2/config/OpenSamlKeyConverters.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package boot.saml2.config;
+
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import org.springframework.boot.context.properties.ConfigurationPropertiesBinding;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.security.saml2.credentials.Saml2X509Credential;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import boot.saml2.config.Saml2SampleBootConfiguration.StringX509Credential;
+import org.bouncycastle.openssl.PEMDecryptorProvider;
+import org.bouncycastle.openssl.PEMEncryptedKeyPair;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JcePEMDecryptorProviderBuilder;
+import org.opensaml.security.x509.X509Support;
+
+import static java.util.Optional.ofNullable;
+import static org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialUsage.DECRYPTION;
+import static org.springframework.security.saml2.credentials.Saml2X509Credential.Saml2X509CredentialUsage.SIGNING;
+
+@Configuration
+class OpenSamlKeyConverters {
```

Comment:
At this point I'd like to defer that inclusion as I am not sure what configuration Spring Boot would adopt.  The sample is just that, a _sample configuration_ and how it can be used.
Spring Boot currently doesn't support conversion of multiple properties into one object, such as `key+passphrase -> private key`.